petab/math/PetabMathExprLexer.g4

```diff
@@ -0,0 +1,35 @@
+// Lexer grammar for PEtab math expressions
+// run `regenerate.sh` to regenerate the lexer
+lexer grammar PetabMathExprLexer;
+
+
+NUMBER          : EXPONENT_FLOAT | INTEGER | POINT_FLOAT | INF;
+INTEGER         : DIGITS ;
+EXPONENT_FLOAT   : (INTEGER | POINT_FLOAT) EXPONENT ;
+POINT_FLOAT      : DIGITS '.' DIGITS ;
+fragment EXPONENT: ('e' | 'E') ('+' | '-')? DIGITS ;
+FLOAT_NUMBER: POINT_FLOAT | EXPONENT_FLOAT;
+fragment DIGITS : [0-9]+ ;
+
+WS      : [ \t\r\n]+ -> skip ;
+TRUE    : 'true' ;
+FALSE   : 'false' ;
+INF     : 'inf' ;
+NAME : [a-zA-Z_][a-zA-Z0-9_]* ;
+OPEN_PAREN : '(' ;
+CLOSE_PAREN : ')' ;
+BOOLEAN_OR : '||' ;
+BOOLEAN_AND : '&&' ;
+GT : '>' ;
+LT : '<' ;
+GTE : '>=' ;
+LTE : '<=' ;
+EQ : '==' ;
+NEQ : '!=' ;
+PLUS : '+' ;
+MINUS : '-' ;
+MUL : '*' ;
+DIV : '/' ;
+HAT: '^';
```

Comment:
Other names are based on operations. I guess this is named `HAT` because `POW` would clash with the function? Could we name this `POWER` instead?

Reply:
For the lexer, I would rather go for character names than associated operations, but yeah, I can try to make it more consistent.